src/background/services/pier-service.ts

```diff
@@ -390,11 +394,58 @@ export class PierService {
             
         } catch (err) {
             console.error(err);
+
+            if (!this.bootingPiers.get(pier.slug).recoveryAttempted)
+                return this.tryAutomatedErrorRecovery(pier);
+
             this.bootingPiers.delete(pier.slug);
             return await this.updatePier(pier.slug, { status: 'errored', lastUsed: (new Date()).toISOString() });
         }
     }
 
+    private async tryAutomatedErrorRecovery(pier: Pier): Promise<Pier | null> {
+        this.bootingPiers.get(pier.slug).recoveryAttempted = true;
+
+        this.db.messageLog.asyncInsert({
+            type: 'out',
+            slug: pier.slug,
+            text: 'Port: attempting automated error recovery.',
+            time: (new Date()).toISOString()
+        });
+
+        let pierPath = this.getSafePath(pier);
+
+        try {
```

Comment:
as far as I'm aware, meld includes pack so we may not need to do this

Reply:
ohhh nice. Couldn't find good documentation about how exactly they work, that's cool. Updated accordingly